tests/phpunit/api/v3/CaseRoleCreationTest.php

```diff
@@ -110,6 +110,50 @@ public function testCaseRolePostProcessingIsNotTriggeredForNonCaseRelatedRelatio
     $this->assertCount(0, $activity['values']);
   }
 
+  /**
+   * Test an existent relationship is not created again.
+   */
+  public function testExistentRelationshipIsNotCreatedAgain() {
+    $caseClient = ContactFabricator::fabricate();
+    $contact = ContactFabricator::fabricate();
+    $relationshipParams = $this->getRelationshipBaseParamsForCaseClient($caseClient);
+
+    $relationshipParams['contact_id_b'] = $contact['id'];
+    $relationshipCreated = civicrm_api3('Relationship', 'create', $relationshipParams);
+    $relationshipCreatedAgain = civicrm_api3('Relationship', 'create', $relationshipParams);
+
+    $this->assertEquals($relationshipCreated['id'], $relationshipCreatedAgain['id']);
+  }
+
+  /**
+   * Test an existent relationship is correctly updated.
+   */
+  public function testExistentRelationshipIsCorrectlyUpdated() {
+    $caseClient = ContactFabricator::fabricate();
+    $firstContact = ContactFabricator::fabricate();
+    $secondContact = ContactFabricator::fabricate();
+    $relationshipParams = $this->getRelationshipBaseParamsForCaseClient($caseClient);
+
+    // Create first relationship.
+    $relationshipParams['contact_id_b'] = $firstContact['id'];
+    $relationshipCreated = civicrm_api3('Relationship', 'create', $relationshipParams);
+    $relationshipCreatedId = array_shift($relationshipCreated['values'])['id'];
+
+    // Update relationship with second contact.
+    $relationshipParams['contact_id_b'] = $secondContact['id'];
+    $relationshipUpdated = civicrm_api3('Relationship', 'create', $relationshipParams);
+    $relationshipUpdated = array_shift($relationshipUpdated['values']);
+
+    // Refresh previous relationship.
+    $relationshipCreated = new CRM_Contact_DAO_Relationship();
+    $relationshipCreated->id = $relationshipCreatedId;
+    $relationshipCreated->find();
+
+    $this->assertNotEquals($relationshipCreated->is_active, $relationshipUpdated['id']);
+    $this->assertEquals(1, $relationshipUpdated['is_active']);
+    $this->assertEquals(0, $relationshipCreated->is_active);
```

Comment:
Not sure why this is inactive. Was it created as inactive?

Reply:
This is what happens on the database level when another same relationship appears. It gets internally marked as inactive, and added an end_date. 
![image](https://user-images.githubusercontent.com/74304572/110822713-ea88ce80-82c3-11eb-8cd6-621a4ea33cf3.png)

The logic is in `CRM_Civicase_Service_CaseRoleCreationPostProcess`.